src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -48,6 +50,41 @@ def _get_db_records(self, use_case_id: UseCaseKey, db_keys: KeyCollection) -> An
 
         return self._table(use_case_id).objects.filter(query_statement)
 
+    def _bulk_create_with_retry(
+        self, table: IndexerTable, new_records: Sequence[BaseIndexer]
+    ) -> None:
+        """
+        With multiple instances of the Postgres indexer running, we found that
+        rather than direct insert conflicts we were actually observing deadlocks
+        on insert. Here we surround bulk_create with a catch for the deadlock error
+        specifically so that we don't interrupt processing or raise an error for a
+        fairly normal event.
+        """
+        retry_count = 0
+        max_retries = 3
+        success = False
+        last_seen_exception: Optional[BaseException] = None
+
+        with metrics.timer("sentry_metrics.indexer.pg_bulk_create"):
+            # We use `ignore_conflicts=True` here to avoid race conditions where metric indexer
+            # records might have be created between when we queried in `bulk_record` and the
+            # attempt to create the rows down below.
+            while not success and retry_count + 1 < max_retries:
+                try:
+                    table.objects.bulk_create(new_records, ignore_conflicts=True)
+                    success = True
```

Comment:
you can just `break` or `return` here instead of keeping a success flag

Reply:
so for the condition on line 83 would you rather I check if retry_count == max_retries - 1?

Reply:
if you `return` then you can run the code at L83 unconditionally